### DISCUSSION TITLE
Duration not working properly with arbitrary value

### DISCUSSION THREAD

**Discussion by maxhotzelt** [tailwindlabs/tailwindcss#13357] [Help] [answered]
According to the Tailwind CSS documentation, the syntax for specifying [arbitrary duration values](https://tailwindcss.com/docs/transition-duration#arbitrary-values) in milliseconds should be `duration-[2000ms]`. However, upon attempting to utilize this syntax, I found that it doesn't seem to apply the transition duration correctly.

For instance, when using `duration-[2000ms]` to specify a duration of 2000 milliseconds, the transition does not apply at all.

![image](https://github.com/tailwindlabs/tailwindcss/assets/164404077/7a37fb0f-becc-4968-b9cc-4e5d633955a8)
![image](https://github.com/tailwindlabs/tailwindcss/assets/164404077/03f9a558-6a42-47c9-bd69-0dce3d221c26)

Furthermore, I wanted to inquire whether the official documentation regarding the usage of custom duration values with milliseconds syntax is incorrect, or if this behavior is indeed a bug within Tailwind CSS. Alternatively, I'm open to help. I may be overlooking something or misusing the feature in some way. Clarification on this matter would be appreciated.

**Comment by alex-krasikau** [comment, marked as answer]
Hey @maxhotzelt!

I tested the arbitrary values implementation on Tailwind Play, and it works well: [Tailwind Play - Arbitrary Values](https://play.tailwindcss.com/WKL0WD9WN5).

I noticed from the screenshots that you are using the `tailwind-animate` plugin. This plugin modifies tailwind transition utilities for animations. There is a bug affecting the use of arbitrary values, which has been reported in several open issues:
-  [Issue 50](https://github.com/jamiebuilds/tailwindcss-animate/issues/50)
-  [Issue 38](https://github.com/jamiebuilds/tailwindcss-animate/issues/38)
-  [Issue 35](https://github.com/jamiebuilds/tailwindcss-animate/issues/35)

To address this problem, I suggest two solutions:

1. Add any values used in the code to the tailwind config:
```js
{
    ...
    theme: {
        extend: {
            transitionDuration: {
                "2000": "2000ms",
            },
        },
    }
    ...
}
```

2. Alternatively, use a fork that prefixes animation utilities with the `anim-` keyword. This may require adjustments in your codebase if you are already using keyframes animations. You can find more details in the [comment](https://github.com/jamiebuilds/tailwindcss-animate/issues/35#issuecomment-1874829634) by the fork author and in the [fork itself](https://github.com/anurag-roy/tailwindcss-animate).
